### PR TITLE
[DWARF] Change to consistently print out abbrev code in .debug_names

### PR DESCRIPTION
**Pull request by ayermolo**
Changed so that when Abbrev code is printed out for entry it is done in the same
way as in Abbrev table.
Once letters are present in a hex number in abbrev table they will be lower case, 
and in the Entry upper case. Which makes FIleCheck Pattern recognition fail.
Example in: llvm/test/tools/llvm-dwarfdump/X86/debug-names-misaligned.s

### CODE REVIEW

**Comment by dwblaikie** [comment]
Could you explain a bit more (maybe a simple example/comparison in the commit description would also be helpful) the motivation for this consistency? They look like they're printing out similarly enough already to be filecheck'd, etc?

**Comment by ayermolo** [comment]
> Could you explain a bit more (maybe a simple example/comparison in the commit description would also be helpful) the motivation for this consistency? They look like they're printing out similarly enough already to be filecheck'd, etc?

Ah right. Difference comes in when letters are involved in hex: upper case vs lower case.
As can be seen in llvm/test/tools/llvm-dwarfdump/X86/debug-names-misaligned.s in the commit.

Which then makes the FileCheck pattern matching fail.
[[ABBREV1:0x[0-9a-f]*]]
[[ABBREV1]]

Will update commit description.

**Comment by felipepiovezan** [comment]
+1 to having the example in the commit description.

I would also suggest rewording the title to drop the "llvm" and be a bit more descriptive. Something like: 
"[DWARF] Use the same abbreviation format for debug_name and debug_abbrev"

**Comment by felipepiovezan** [comment]
Note that only the PR description was updated, the commit message is still the old one

**Comment by ayermolo** [comment]
> Note that only the PR description was updated, the commit message is still the old one

Commit message updated. Something else came up, after I update PR description. :)

**Comment by dwblaikie** [comment]
Thanks for explaining. Changing the formatting layout of the name dumping seems not ideal - scoping the attributes makes it read as though they're part of an appreviation, which they aren't (they're part of the entry, using the abbrev as the encoding/schema).

So perhaps we can change only the case of the hex characters, rather than other aspects of the rendering.

It looks like the root cause of the divergence is that `raw_ostream::write_hex` explicitly uses lower case, and `llvm::operator<<(raw_ostream&, const HexNumber&)` gets implicit upper case.

Perhaps the latter should use the former and have the same behavior? Probably requires a bit of cleanup to other tests/usage, but hopefully it's feasible...

**Comment by ayermolo** [comment]
> Thanks for explaining. Changing the formatting layout of the name dumping seems not ideal - scoping the attributes makes it read as though they're part of an appreviation, which they aren't (they're part of the entry, using the abbrev as the encoding/schema).
> 
> So perhaps we can change only the case of the hex characters, rather than other aspects of the rendering.
> 
> It looks like the root cause of the divergence is that `raw_ostream::write_hex` explicitly uses lower case, and `llvm::operator<<(raw_ostream&, const HexNumber&)` gets implicit upper case.
> 
> Perhaps the latter should use the former and have the same behavior? Probably requires a bit of cleanup to other tests/usage, but hopefully it's feasible...

Ah... I see your point about scoping. Let me change.

**Review by dwblaikie — APPROVED** [review]
